### PR TITLE
chore(login): improve login page

### LINKED ISSUE #146: Improve login page
The applications login page is first interface a user will interact with in the application.  Currently, the error handling and validation messages could all be improved.  Some improvements are listed below, and any comments will be added to this list.
##### Improvements
- [ ] Ensure that error messages are returned from the server and translated at the bottom of the login form (in a `class="help-block"`):
  1. "Incorrect username or password"
  2. "No projects registered"  (installation error)
  3. "No enterprises"  (installation error)
  4. "No connection to the server" (network error)
- [ ] Turn `autocomplete`, `autocorrect`, etc, off on the username `<input>`
- [ ] Show "help" text if the user fails to login three or more times
  1. "Enter your username into this field."
  2. "Enter your password into this field."
  3. "Forgot your username or password?  Ask the system administrator."

### PR DESCRIPTION
**Pull request by jniles**
This commit improves the login page in the following ways:
1. Removes `<alert>`s in favor of cleaner error messages
2. Responds appropriately to both form and server errors, displaying translatable text on the page.
3. Turns off `autocomplete` madness on the username form.
4. Gives a hint if the user attempts to sign in incorrectly too many times.

Closes #146.

### CODE REVIEW

File: client/src/js/services/authinjector.js [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
-function AuthInjector($rootScope, $q, $location, Session) {
+AuthInjectorFactory.$inject = [
+  '$rootScope','$q',  '$location', 'SessionService'
+];
+
+function AuthInjectorFactory($rootScope, $q, $location, Session) {
   return {
     responseError : function (res) {
-      console.log('[INTERCEPTOR] Received HTTP error', res.status, res.data);
+      console.log('[INTERCEPTOR] Received HTTP error', res.status);
```
Comment (DedrickEnc) [inline client/src/js/services/authinjector.js:11]: We will continue to keep this console.log?

File: client/src/js/services/authinjector.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
 angular.module('bhima.services')
-.factory('AuthInjectorFactory', ['$rootScope','$q',  '$location', 'SessionService', AuthInjector]);
+.factory('AuthInjectorFactory', AuthInjectorFactory);
```
Comment (DedrickEnc) [inline client/src/js/services/authinjector.js:2]: Why do you need a factory?

File: client/src/partials/login/login.js [hunk 3]
```diff
@@ -14,12 +14,19 @@ function LoginController($scope, $location, $http, $timeout, AppCache, appstate,
   // the is the same as the SettingsContoller
   var cache = AppCache('preferences');
 
+  // local variable count to
```
Comment (DedrickEnc) [inline client/src/partials/login/login.js:17]: This comment is funny

**Comment by DedrickEnc** [comment]
@jniles 
This PR looks good to me, it can be merged in any time, however I asked some questions to be sure about some choice you made. If you are sure about your choices, I will merge it in.

File: client/src/js/services/authinjector.js (hunk 1 above)
Comment (jniles) [inline client/src/js/services/authinjector.js:11]: It is helpful to know the error code that came back in development.  It should be removed in production code though.

File: client/src/js/services/authinjector.js (hunk 2 above)
Comment (jniles) [inline client/src/js/services/authinjector.js:2]: This is to make an $http interceptor, which must return an object. A service would also work, I am pretty sure, but have not tested it.

**Comment by jniles** [comment]
What questions do you have?

**Comment by DedrickEnc** [comment]
You already answered to them.

**Pull request merged**